test/Microsoft.AspNet.Mvc.ModelBinding.Test/Binders/FormFileModelBinderTest.cs

```diff
@@ -0,0 +1,124 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNet.Http;
+using Microsoft.AspNet.PipelineCore;
+using Microsoft.AspNet.PipelineCore.Collections;
+using Moq;
+using Xunit;
+
+namespace Microsoft.AspNet.Mvc.ModelBinding
+{
+    public class FormFileModelBinderTest
+    {
+        [Fact]
+        public async Task FormFileModelBinder_ExpectMultipleFiles_BindSuccessful()
+        {
+            // Arrange
+            var httpContext = GetMockHttpContext(GetMockFormCollection());
+            var bindingContext = GetBindingContext(typeof(IEnumerable<IFormFile>), httpContext);
+            var binder = new FormFileModelBinder();
+
+            // Act
+            var result = await binder.BindModelAsync(bindingContext);
+
+            // Assert
+            Assert.True(result);
+            var files = bindingContext.Model as IList<IFormFile>;
+            Assert.Equal(files.Count, 2);
+        }
+
+        [Fact]
+        public async Task FormFileModelBinder_ExpectSingleFile_BindFirstFile()
+        {
+            // Arrange
+            var httpContext = GetMockHttpContext(GetMockFormCollection());
+            var bindingContext = GetBindingContext(typeof(IFormFile), httpContext);
+            var binder = new FormFileModelBinder();
+
+            // Act
+            var result = await binder.BindModelAsync(bindingContext);
+
+            // Assert
+            Assert.True(result);
+            var file = bindingContext.Model as IFormFile;
+            Assert.Equal(file.ContentDisposition,
+                        "form-data; name=\"file\"; filename=\"file1.txt\"");
+        }
+
+        [Fact]
+        public async Task FormFileModelBinder_ReturnsNull_WhenNoFilePosted()
+        {
+            // Arrange
+            var httpContext = GetMockHttpContext(GetEmptyFormCollection());
+            var bindingContext = GetBindingContext(typeof(IFormFile), httpContext);
+            var binder = new FormFileModelBinder();
+
+            // Act
+            var result = await binder.BindModelAsync(bindingContext);
+
+            // Assert
+            Assert.True(result);
+            Assert.Null(bindingContext.Model);
+        }
```

Comment:
what about when the names don't match?

what about missing/unparsable content disposition?